src/test/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/InntektsmeldingRepositoryTest.kt

```diff
@@ -1,23 +1,152 @@
+        val forespoerselId = UUID.randomUUID().toString()
+        val inntektsmeldingJson = buildInntektsmeldingJson(forespoerselId)
+        transaction(db) {
+            val org = "123456789"
+            val sykmeldtFnr = "10107400090"
+            val innsendtDato = LocalDateTime.of(2023, 1, 1, 0, 0)
+            repository.opprett(
+                inntektsmeldingJson,
+                org,
+                sykmeldtFnr,
+                innsendtDato,
+                forespoerselId,
+            )
+
+            val result =
+                repository.hent(
+                    org,
+                    InntektsmeldingRequest(
+                        fnr = sykmeldtFnr,
+                        foresporselid = forespoerselId,
+                        datoFra = innsendtDato.minusDays(1),
+                        datoTil = innsendtDato.plusDays(1),
+                    ),
+                )
+
+            assertEquals(1, result.size)
+            assertEquals(org, result[0].orgnr)
+            assertEquals(sykmeldtFnr, result[0].fnr)
+            assertEquals(forespoerselId, result[0].foresporselid)
+            rollback()
+        }
+    }
+
     @Test
-    fun opprett() {
-        val db = Database.init()
-        val repo = InntektsmeldingRepository(db)
-        val dok = """{"test":"test"}"""
-        val orgnr = "123"
-        val fnr = "321"
-        val forespoerselID = "1234"
-        repo.opprett(dok, orgnr, fnr, forespoerselID)
-        val inntektsmelding = repo.hent(orgnr)[0]
-        inntektsmelding.foresporselid shouldBe forespoerselID
-        inntektsmelding.orgnr shouldBe orgnr
-        inntektsmelding.fnr shouldBe fnr
-        inntektsmelding.dokument.toString() shouldBe dok
+    fun `hent should return list av inntektsmeldinger by orgNr and request with no match`() {
+        transaction(db) {
+            val org1 = "123456789"
+            val sykmeldtFnr1 = "10107400090"
+            val innsendtDato1 = LocalDateTime.of(2023, 1, 1, 0, 0)
+            val forespoerselId1 = UUID.randomUUID().toString()
+            generateTestData(repository, org1, sykmeldtFnr1, innsendtDato1, forespoerselId1)
+
+            val org2 = "987654321"
+            val sykmeldtFnr2 = "10107400091"
+            val innsendtDato2 = LocalDateTime.of(2023, 1, 2, 0, 0)
+            val forespoerselId2 = UUID.randomUUID().toString()
+            generateTestData(repository, org2, sykmeldtFnr2, innsendtDato2, forespoerselId2)
+
+            val result =
+                repository.hent(
+                    "987654321",
+                    InntektsmeldingRequest(
+                        fnr = null,
+                        foresporselid = null,
+                        datoFra = null,
+                        datoTil = null,
+                    ),
+                )
+
+            assertEquals(1, result.size)
+            rollback()
+        }
+    }
+
+    private fun generateTestData(
```

Comment:
forslag: ikke sende inn repository her - dele opp litt i to steg ala repository.opprett(genererTestIM())